src/components/marker.js

```diff
@@ -58,18 +49,33 @@ export default class Marker extends DraggableControl {
   static propTypes = propTypes;
   static defaultProps = defaultProps;
 
-  _render() {
-    const {className, longitude, latitude, offsetLeft, offsetTop} = this.props;
+  _getPosition(): [number, number] {
+    const {longitude, latitude, offsetLeft, offsetTop} = this.props;
     const {dragPos, dragOffset} = this.state;
 
-    const [x, y] = dragPos ?
-      this._getDraggedPosition(dragPos, dragOffset) :
-      this._context.viewport.project([longitude, latitude]);
+    // If dragging, just return the current drag position
+    if (dragPos) {
+      return this._getDraggedPosition(dragPos, dragOffset);
+    }
+
+    // Otherwise return the projected lat/lng with offset
+    let [x, y] = this._context.viewport.project([longitude, latitude]);
+    x += offsetLeft;
+    y += offsetTop;
+    return [x, y];
+  }
+
+  _render() {
+    const {className, draggable} = this.props;
+    const {dragPos} = this.state;
+
+    const [x, y] = this._getPosition();
 
     const containerStyle = {
       position: 'absolute',
-      left: x + offsetLeft,
-      top: y + offsetTop
+      left: x,
+      top: y,
+      cursor: draggable ? (dragPos ? 'grabbing' : 'grab') : 'auto'
```

Comment:
@Pessimistress This isn't perfect because sometimes the mouse moves faster than the marker, but it seems to work reasonably well. Let me know if you there's a better way of achieving this.

Reply:
You can set the cursor style on the entire canvas to fix this. Without doing it the "React way" you can access the root element through the pointer events...

Reply:
Gotcha. Sounds like something that might require its own PR, and the small change here should be a decent improvement for now.